trace-dispatcher/src/Cardano/Logging/ConfigurationParser.hs

```diff
@@ -95,57 +151,74 @@ parseRepresentation bs = transform (decodeEither' bs)
           (traceOptionPeerFrequency cr)
           (traceOptionResourceFrequency cr)
 
-data ConfigRepresentation = ConfigRepresentation {
-    traceOptions                :: OptionsRepresentation
-  , traceOptionForwarder        :: Maybe TraceOptionForwarder
-  , traceOptionNodeName         :: Maybe Text
-  , traceOptionPeerFrequency     :: Maybe Int
-  , traceOptionResourceFrequency :: Maybe Int
-  }
-  deriving (Eq, Ord, Show, Generic)
-
-type OptionsRepresentation = Map.Map Text ConfigOptionRep
-
-instance AE.FromJSON ConfigRepresentation where
-    parseJSON (Object obj) = ConfigRepresentation
-                           <$> obj .: "TraceOptions"
-                           <*> obj .:? "TraceOptionForwarder"
-                           <*> obj .:? "TraceOptionNodeName"
-                           <*> obj .:? "TraceOptionPeerFrequency"
-                           <*> obj .:? "TraceOptionResourceFrequency"
-    parseJSON _ = mempty
+      -- | Convert from external to internal representation
+    toConfigOptions :: ConfigOptionRep -> [ConfigOption]
+    toConfigOptions ConfigOptionRep {..} =
+      case severity of
+        Nothing -> []
+        Just sev -> [ConfSeverity sev]
+      ++
+      case detail of
+        Nothing -> []
+        Just dtl -> [ConfDetail dtl]
+      ++
+      case backends of
+        Nothing -> []
+        Just bcks -> [ConfBackend bcks]
+      ++
+      case maxFrequency of
+        Nothing -> []
+        Just lim -> [ConfLimiter lim]
+
```

Comment:
Could you use sth. like `maybeToList (ConfDetail <$> maxDetail)` here?

Reply:
Clever, I changed it